src/cpp/shared/campaign/campaign.cpp

```diff
@@ -400,60 +454,243 @@ void ExpCampaign::ShowInfo(int inGame, int unused) {
   }
 }
 
-void ExpCampaign::ReplaySmacker() {
-  if(this->campaignID < MAX_CAMPAIGNS) {
-    switch(this->campaignID) {
-      case 0:
-        this->ReplaySmacker1();
+std::vector<std::string> xCampaignAwards = {
+  "Elven alliance",
+  "Breastplate",
+  "Wood bonus",
+  "Helmet",
+  "Kraeger Defeated",
+  "Battle Garb",
+  "Wayward son",
+  "Uncle Ivan",
+  "Legendary Sceptre",
+  "Guardian Spell",
+  "Sphere of Negation"
+};
+
+std::vector<std::string> xCampaignShortSkillLevel = {
+  "Bas.",
+  "Adv.",
+  "Exp."
+};
+
+void ExpCampaign::UpdateInfo(int drawWindow) {
```

Comment:
What exactly does this do, and when is it called?

In particular, why didn't it need to be changed for the Cyborg campaign previously?

Reply:
Previously it was using original game array accesses. Now it uses our own (std::map items mostly)